src/rebar3_hex_publish.erl

```diff
@@ -174,38 +135,161 @@ format_error(Reason) ->
     rebar3_hex_error:format_error(Reason).
 
 %% ===================================================================
-%% Public API
+%% Private
 %% ===================================================================
 
-publish(App, HexConfig, State) ->
-    Name = rebar_app_info:name(App),
 
-    Version = rebar_app_info:original_vsn(App),
-    ResolvedVersion = rebar_utils:vcs_vsn(App, Version, State),
-    {application, _, AppDetails} = rebar3_hex_file:update_app_src(App, ResolvedVersion),
+%% ===================================================================
+%% Publish package only operations
+%% ===================================================================
+handle_task(#{args := #{task := package, app := undefined}, multi_app := true}) ->
+    ?RAISE({app_switch_required, "--app required when publishing with the package argument in a umbrella"});
```

Comment:
Rationale : If you're only wanting to publish a package or docs (not both) then you probably only want to do it on a single app basis, but would defer to @tsloughter